src/transport.ts

```diff
@@ -357,24 +357,28 @@ export class ReconnectableTransport extends EventEmitter implements ITransport {
   }
 
   private isOnlinePromise(mode: ReconnectionMode) {
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
       const checkSocket = new WebSocket(this.uaOptions.transportOptions.wsServers, 'sip');
 
       const handlers = {
         onError: e => {
           log.debug(e, this.constructor.name);
+
           checkSocket.removeEventListener('open', handlers.onOpen);
-          // In the case that mode is BURST, throw an error which can be
-          // catched by pRetry.
+          checkSocket.removeEventListener('error', handlers.onError);
+
+          // In the case that mode is BURST, reject the promise which can be
+          // caught by pRetry.
           if (mode === ReconnectionMode.BURST) {
-            throw new Error('it broke woops');
+            reject('reject to trigger pRetry that tries to reconnect to sip server');
```

Comment:
```suggestion
            reject('Connection to sip server broke, try to reconnect');
```

Or something like that :)

Reply:
This is an "internal error" as this specific error should be caught and it should never see the light of day but I do like your message more!